utilities/transactions/write_prepared_txn.h

```diff
@@ -32,8 +32,34 @@
 
 namespace rocksdb {
 
+class WritePreparedTxn;
+class WriteUnpreparedTxn;
 class WritePreparedTxnDB;
 
+class WritePreparedTxnReadCallback : public ReadCallback {
+ public:
+  WritePreparedTxnReadCallback(WritePreparedTxnDB* db, SequenceNumber snapshot,
+                               SequenceNumber min_uncommitted)
+      : txn_(nullptr),
+        db_(db),
+        snapshot_(snapshot),
+        min_uncommitted_(min_uncommitted) {}
+
+  // Will be called to see if the seq number accepted; if not it moves on to the
+  // next seq number.
+  virtual bool IsCommitted(SequenceNumber seq) override;
+
+  virtual SequenceNumber MaxVisibleSequenceNumber() override;
+
+ protected:
+  friend class WriteUnpreparedTxnDB;
+
+  WriteUnpreparedTxn* txn_;
```

Comment:
Why is that there is pointer to "unprepared" from "prepared" callback?

Reply:
See my comment in `WritePreparedTxnReadCallback::IsCommitted`.